src/modules/admin/components/forms/FormPreview.tsx

```diff
@@ -102,23 +109,58 @@ const FormPreview = () => {
     );
   }, [formDefinition, formValues, initialValues, localConstants, toggleValue]);
 
+  const [confirmOpen, setConfirmOpen] = useState(false);
+  const [publishForm, { loading: publishLoading, error: publishError }] =
+    usePublishFormMutation();
+
+  const onPublish = useCallback(() => {
+    if (!formDefinition) return;
+
+    publishForm({
+      variables: {
+        id: formDefinition.id,
+      },
+      onCompleted: () => {
+        cache.evict({ id: `FormIdentifier:${formDefinition?.identifier}` });
+      },
+    });
+  }, [publishForm, formDefinition]);
+
   if (loading && !formDefinition) return <Loading />;
   if (error) throw error;
+  if (publishError) throw publishError;
```

Comment:
I'd like to confirm my judgment call that there's no need to use `ErrorState` and `apolloError` here, because any errors here would be bugs (as opposed to user input validation), does that sound right?